test/user.js

```diff
@@ -0,0 +1,125 @@
+'use strict'
+
+const { expect } = require('chai')
+const request = require('supertest')
+const sinon = require('sinon')
+const User = require('../models/User')
+const app = require('..')
+
+describe('test user controller', () => {
+  let authorization
+
+  before(() => {
+    return request(app)
+      .post('/login')
+      .send({ user: 'admin', password: '123456' })
+      .then(res => {
+        authorization = `Bearer ${res.body.access_token}`
+      })
+  })
+
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('should GET /users 200', () => {
+    sinon.stub(User, 'find').resolves([])
```

Comment:
Some teams think that trying to mock database is unwise/impractical for test cases, can you compare the benefit & cost of mocking database?

Reply:
Mocking is simple and useful in this scenario, no need to configure a test database, no dirty data will be generated.
When testing some complex functions, use a real database would be simple and convincing.